src/coreclr/jit/helperexpansion.cpp

```diff
@@ -795,22 +795,17 @@ PhaseStatus Compiler::fgExpandStaticInit()
         return result;
     }
 
-    if (opts.OptimizationDisabled())
-    {
-        JITDUMP("Optimizations aren't allowed - bail out.\n")
-        return result;
-    }
+    // Always expand for NativeAOT, see
+    // https://github.com/dotnet/runtime/issues/68278#issuecomment-1543322819
```

Comment:
I'm going to look into deleting the assembly helpers that have been doing until now - this comment can be replaced with "native AOT doesn't have helpers to do this". This simplifies the product as a side effect of fixing the issue with RVA statics + cctors.

Reply:
I'll update these comments in my next PR for NAOT (guarded devirt) this week to avoid spinning CI for comments updates if you don't mind 🙂